lib/iris/fileformats/pp.py

```diff
@@ -1591,20 +1598,31 @@ def _interpret_fields(fields):
                 (field.lbuser[3] % 1000) == 30:
             land_mask = field
 
+        apply_landmask = None
+
         # Handle land compressed data payloads,
         # when lbpack.n2 is 2.
         if (field.raw_lbpack // 10 % 10) == 2:
             if land_mask is None:
                 landmask_compressed_fields.append(field)
+                # Land-masked fields have their size+shape defined by the
+                # reference landmask field, so we can't yield them if they
+                # are encountered *before* the landmask.
+                # In that case, defer them, and process them all afterwards at
+                # the end of the file.
                 continue
 
-            # Land compressed fields don't have a lbrow and lbnpt.
+            # Landmask-compressed fields don't have an lbrow and lbnpt.
```

Comment:
@pp-mo Minor point - Just to match the terminology of your previous [comment](https://github.com/SciTools/iris/pull/3255/files#diff-066f828420b27bfd2a03868477b98415R1608), could you change this to `Land-masked compressed fields..`?